### ISSUE TITLE
Apache Sling performance with OpenJ9

### ISSUE THREAD

**Issue by babsingh** [eclipse/openj9#11194]
Blog: https://blogs.perficient.com/2020/08/19/apache-sling-jvm-performance-comparison/

According to the above blog, OpenJ9, with respect to Hotspot, takes 
1. **87% longer to startup** [negative]
2. **35% longer to install** [negative]
3. **14% less throughout** [negative]
4. **55% less average memory** [positive]

**Comment by DanHeidinga** [comment]
It looks like it's doing an install of the JDK and then the app and then applying load.  Which will be filling the default SCC but never taking advantage of it.  So a cold run only.  I wonder how much improvement they'd see in startup time by starting the app and doing their measurements on a second startup.

**Comment by pshipton** [comment]
@vijaysun-omr fyi

**Comment by mpirvu** [comment]
Scripts here https://github.com/PerficientDigital/sling-jvm-comparison/blob/master/mount/test.sh#L63-L74 show the start-up is repeated 3 times

**Comment by babsingh** [comment]
> the start-up is repeated 3 times, so the default SCC should be used.

There are three `TRY` attempts in case of failures. There is a `break` statement after the first successful start up. So, there is only `ONE` successful start-up.

**Comment by mpirvu** [comment]
Good catch. I missed that. It's hard to believe though someone would do just a single run to measure performance.

**Comment by vijaysun-omr** [comment]
Is this something we want to suggest directly to the blogger to try ? i.e. at least see if the default SCC can be used.

**Comment by mpirvu** [comment]
There is an outer loop here https://github.com/PerficientDigital/sling-jvm-comparison/blob/master/mount/test.sh#L48-L59
which repeats each test 4 times, so there must be some default SCC, unless some other script deletes it (I couldn't find signs of that though).

I performed some start-up experiments of Apache Sling on a skylake based desktop and I can say that the start-up time varies a lot. Averages of 30 runs with outliers excluded:

OpenJ9 default SCC (no options): 5.9 sec
OpenJ9 no SCC: 7.1 sec
OpenJ9 explicit SCC: 4.9 sec
HotSpot: 5.9 sec

One weird thing is that my absolute numbers are way smaller than the range of values in the blog: 27-50 sec

**Comment by mpirvu** [comment]
Seeing this VirtualBox machine definition in the gitgub repo for the test
```
Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/bionic64"
  config.vm.provider "virtualbox" do |v|
    v.memory = 4096
    v.cpus = 2
  end
  config.vm.synced_folder "../mount/", "/opt/mount"
  config.vm.provision "shell", inline: "/bin/bash /opt/mount/init.sh"
end
```
I also performed experiments with the JVM pinned to two HW threads. Again, I do not see OpenJ9 being slower than HotSpot when no options are provided:
```
HotSpot: 		 10872
OpenJ9 no SCC: 	 11957
OpenJ9 default:  10235
OpenJ9 SCC: 	  7210
```
